mo_gymnasium/wrappers/vector/wrappers.py

```diff
@@ -101,6 +112,249 @@ def step(self, actions: ActType) -> Tuple[ObsType, ArrayType, ArrayType, ArrayTy
             np.copy(self._truncations),
             infos,
         )
+    
+def _mo_async_worker(
+    index: int,
+    env_fn: callable,
+    pipe: Connection,
+    parent_pipe: Connection,
+    shared_memory: Union[Array, Dict[str, Any], Tuple[Any, ...]],
+    error_queue: Queue,
+    autoreset_mode: AutoresetMode,
+):
+    env = env_fn()
+    observation_space = env.observation_space
+    action_space = env.action_space
+    reward_space = env.unwrapped.reward_space
+    autoreset = False
+    observation = None
+
+    parent_pipe.close()
+
+    try:
+        while True:
+            command, data = pipe.recv()
+
+            if command == "reset":
+                observation, info = env.reset(**data)
+                if shared_memory:
+                    write_to_shared_memory(
+                        observation_space, index, observation, shared_memory
+                    )
+                    observation = None
+                    autoreset = False
+                pipe.send(((observation, info), True))
+            elif command == "reset-noop":
+                pipe.send(((observation, {}), True))
+            elif command == "step":
+                if autoreset_mode == AutoresetMode.NEXT_STEP:
+                    if autoreset:
+                        observation, info = env.reset()
+                        reward, terminated, truncated = np.zeros(reward_space.shape[0], dtype=np.float32), False, False
```

Comment:
Would be useful for maintainance to annotate with a comment which lines have been modified compared to the original Gymnasium